relay-filter/src/config.rs

```diff
@@ -192,6 +196,86 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericFilterConfig {
+    /// Specifies whether this filter is enabled.
+    pub is_enabled: bool,
+    /// The condition for the filter.
+    pub condition: Option<RuleCondition>,
+}
+
+impl GenericFilterConfig {
+    /// Returns true if the filter is not enabled or the condition is not supported.
+    pub fn is_empty(&self) -> bool {
+        !self.is_enabled || matches!(self.condition, Some(RuleCondition::Unsupported))
+    }
+}
+
+/// Configuration for generic filters.
+///
+/// We use a vector in order to guarantee consistent total ordering of filters that is required
+/// by the matching algorithm.
+#[derive(Clone, Debug, Default)]
+pub struct GenericFiltersConfig(pub Vec<(String, GenericFilterConfig)>);
```

Comment:
Yeah, that is a good idea, I can definitely do it. If the new object is also a struct without named fields, it will be equivalent to the current `JSON` representation, I suppose. Will try it out!

Reply:
@iker-barriocanal I get the idea, but in this case it is overkill IMO. Every generic filter will be configured independently of other generic filters, so the chance that they will ever have shared properties is low. If they do, we can always add additional config one level higher (`{"filters": {"generic": ..., "genericSharedProps": ...}`).